sw/host/ot_certs/src/asn1/codegen.rs

```diff
@@ -487,9 +487,13 @@ impl Builder for Codegen<'_> {
         let str_type = str_type.codestring();
         match val {
             Value::Literal(x) => {
-                // NOTE: we are not using a constant for the size of the string because technically there is no way
-                // to know what the `char` type is and therefore what the length of this string is for the C compiler.
-                self.push_str_with_indent(&format!("RETURN_IF_ERROR(asn1_push_string(&state, {str_type}, \"{x}\", strlen(\"{x}\")));\n"));
+                // All strings are either ASCII or UTF-8 so if we assume that a char corresponds to a byte (an assumption that should hold
```

Comment:
FYI Rust's `str::len` always returns bytes, so we don't have to make this assumption

Reply:
Sorry, the text was not clear apparently: the assumption is on the C side. The C standard allows for implementations where `char` is not an 8-bit byte and I am not sure that the standard guarantees anything about the encoding of literals, it could be something else than UTF8 in principle, and in that case the length in bytes would not match the number of chars on the C side. In an ideal world I should maybe have avoided `char *` in the API for strings?

Reply:
I don't think this is something to worry about at all, and we'd better not have that comment.

We're targeting RISC-V and it's guaranteed to be 8-bit. We're not targeting PDP, having this comment just adds noise without much benefit.

Reply:
> Sorry, the text was not clear apparently: the assumption is on the C side.

Ah I did misunderstand, sorry - I thought you were talking about Rust's `char` which can be multiple bytes which we can ignore because `len` gives us the number of bytes and not chars.

> In an ideal world I should maybe have avoided char * in the API for strings

Yeah, this would make sense for complete compatibility with the Rust side. As @nbdd0121 mentions we can probably ignore this for our chip where we know `char` will always be `uint8_t`.